src/post/post.repository.knex.ts

```diff
@@ -155,21 +151,34 @@ export class KnexPostRepository {
                     }
                 }
 
-                if (potentiallyNewReposts.length > 0) {
-                    const { count, accountIds } =
-                        await this.insertRepostsIgnoringDuplicates(
-                            post,
-                            potentiallyNewReposts,
-                            transaction,
-                        );
-
-                    repostAccountIds = accountIds;
+                if (repostsToAdd.length > 0 || repostsToRemove.length > 0) {
+                    const { insertedRepostsCount, accountIdsInserted } =
+                        repostsToAdd.length > 0
+                            ? await this.insertRepostsIgnoringDuplicates(
+                                  post,
+                                  repostsToAdd,
+                                  transaction,
+                              )
+                            : { insertedRepostsCount: 0, accountIdsInserted: [] };
+
+                    const removedRepostsCount =
+                        repostsToRemove.length > 0
+                            ? await this.removeReposts(
+                                  post,
+                                  repostsToRemove,
+                                  transaction,
+                              )
+                            : 0;
+
+                    repostAccountIds = accountIdsInserted.filter(
```

Comment:
We use these repostAccountIds to emit `PostRepostedEvent`, do we also need something like `PostDeRepostedEvent`?

Reply:
Yeah i think so, although I think that falls under the umbrella of deleting content 🤔 (as we would need to remove the `post` from the `feed` on receipt of this event). Not sure if we want to do this now or later?